### PR TITLE
Experience/6495: open api/validate endpoint to unauthenticated users

### PR DESCRIPTION
**Pull request by ahay-agile6**
This PR opens the validate endpoint to unauthenticated users, adds "generic" senders for the upload-covid-19 scehme, and makes some validation checks, such as PAYLOAD_MAX_BYTES, more flexible between different endpoints and serializers.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. *Include steps to test these changes*

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/serializers/Hl7Serializer.kt [hunk 1]
```diff
@@ -397,12 +397,16 @@ class Hl7Serializer(
         val mapping = convertBatchMessagesToMap(messageBody, schema, sender = sender)
         val mappedRows = mapping.mappedRows
 
+        // Generate the action log
+        val actionLogs = ActionLogger()
+
+        // throw an error if the max items limit is exceeded
+        SerializerValidation.throwMaxItemsError(mappedRows.size, actionLogs)
```
Comment (ahay-agile6) [inline prime-router/src/main/kotlin/serializers/Hl7Serializer.kt:404]: i am not sure we want to keep this. this is an example of the shared max size validator.
we can make this even more configurable by making the validator take a max size parameter. 
that way each Serializer can have its own max size without affecting the other.
right now, we're simply using the defaults that were in place before

File: prime-router/settings/organizations.yml [hunk 2]
```diff
@@ -4193,3 +4193,23 @@
       schemaName: "upload-covid-19"
       keys: null
       processingType: "sync"
+
+- name: "validation-service-senders"
```
Comment (ahay-agile6) [inline prime-router/settings/organizations.yml:4197]: we can change the name of this. lol :laughing:

File: prime-router/src/main/kotlin/azure/HttpUtilities.kt [hunk 3]
```diff
@@ -210,7 +210,7 @@ class HttpUtilities {
          * Do a variety of checks on payload size.
          * Returns a Pair (http error code, human readable error message)
          */
-        fun payloadSizeCheck(request: HttpRequestMessage<String?>) {
+        fun payloadSizeCheck(request: HttpRequestMessage<String?>, maxPayloadSize: Long = PAYLOAD_MAX_BYTES) {
```
Comment (ahay-agile6) [inline prime-router/src/main/kotlin/azure/HttpUtilities.kt:213]: making `PAYLOAD_MAX_BYTES` configurable for whoever needs to call this. it defaults to the Report.kt constant

**Comment by stephenkao** [comment]
We've already allowed the validation endpoint to be public, and we can reassess if we need to add any more restrictions like payload size later on.  Closing!

**Pull request closed without merging**